src/v/redpanda/admin/security.cc

```diff
@@ -773,3 +780,36 @@ admin_server::get_role_handler(std::unique_ptr<ss::http::request> req) {
     }
     co_return ss::json::json_return_type(j_res);
 }
+
+ss::future<ss::json::json_return_type>
+admin_server::update_role_handler(std::unique_ptr<ss::http::request> req) {
+    if (need_redirect_to_leader(model::controller_ntp, _metadata_cache)) {
+        // In order that we can do a reliably ordered validation of
+        // the request (and drop no-op requests), run on controller leader;
```

Comment:
We'd need an internal RPC to forward to the controller leader node, right? How do we decide when to forward internally with an RPC vs relying on client forwarding?

Reply:
>We'd need an internal RPC

Yeah, I believe so.

>How do we decide

No clue, honestly.  w/in the admin API, the redirect seems more general to me, but if there's a hidden cost to doing things this way, we should make a determination. 

Probably not for this PR though (since this usage is fairly prevalent across `v1/security/*`

Reply:
> We'd need an internal RPC to forward to the controller leader node, right?

Yes.

> How do we decide when to forward internally with an RPC vs relying on client forwarding?

Good question. I wasn't actually aware of how this 307 worked. Should be ok.